lib/arith.gd

```diff
@@ -966,8 +966,14 @@ DeclareCategoryCollections("IsZDFRECollection");
 ##  <Filt Name="IsMatrix" Arg='obj' Type='Category'/>
 ##
 ##  <Description>
-##  A <E>matrix</E> is a list of lists of equal length whose entries lie in a
-##  common ring.
+##  By convention <E>matrix</E> is a list of lists of equal length whose
+##  entries lie in a common ring.
+##  <P/>
+##  For technical reasons laid out at the top of Chapter <Ref Chap="Matrices"/>,
+##  the filter <Ref Filt="IsMatrix"/> is a synonym for a table of ring elments,
+##  (see <Ref Filt="IsTable"/> and <Ref Filt="IsRingElement"/>)
+##  If necessary, <Ref Prop="IsRectangularTable"/> can be used to test whether
+##  an object is a matrix manually.
```

Comment:
While this is already an improvment, I think we can still do better, by explicitly stating some consequences of this technical point -- as it is, most readers will be none the wiser after reading this. So how about adding something like "This means that sometimes IsMatrix returns true for things were it really shouldn't, e.g. ..."